checkov/common/util/secrets.py

```diff
@@ -133,20 +133,28 @@ def omit_secret_value_from_line(secret: str, line_text: str) -> str:
 
 def omit_secret_value_from_checks(check: BaseCheck, check_result: dict[str, CheckResult] | _CheckResult,
                                   entity_code_lines: list[tuple[int, str]],
-                                  entity_config: dict[str, Any] | ParameterAttributes | ResourceAttributes) -> \
+                                  entity_config: dict[str, Any] | ParameterAttributes | ResourceAttributes,
+                                  resource_attributes_to_omit: ResourceAttributesToOmit | None = None) -> \
         list[tuple[int, str]]:
+    secrets = set()  # a set, to efficiently avoid duplicates in case the same secret is found in the following conditions
+    censored_code_lines = []
+
     if CheckCategories.SECRETS in check.categories and check_result.get('result') == CheckResult.FAILED:
-        censored_code_lines = []
-        secrets = [str(secret) for key, secret in entity_config.items() if key.startswith(f'{check.id}_secret')]
-        if not secrets:
-            logging.debug(f"Secret was not saved in {check.id}, can't omit")
-            return entity_code_lines
-
-        for idx, line in entity_code_lines:
-            censored_line = omit_multiple_secret_values_from_line(secrets, line)
-            censored_code_lines.append((idx, censored_line))
-    else:
-        censored_code_lines = entity_code_lines
+        secrets.update([str(secret) for key, secret in entity_config.items() if key.startswith(f'{check.id}_secret')])
+
+    if resource_attributes_to_omit and check.entity_type in resource_attributes_to_omit and \
+            resource_attributes_to_omit.get(check.entity_type, None) in entity_config:
+        secret = entity_config.get(resource_attributes_to_omit.get(check.entity_type, {}), [])  # type:ignore[arg-type]
+        if isinstance(secret, list) and len(secret) != 0:
```

Comment:
```suggestion
        if isinstance(secret, list) and len(secret) != 0:
```
```suggestion
        if isinstance(secret, list) and secret:
```
not empty check, without using `len` is faster